containers/withAdminContainer.ts

```diff
@@ -0,0 +1,45 @@
+import { Context } from '../@types/helpers'
+import { isAdmin, isAdminOrThrow } from '../helpers/isAdmin'
+import _ from 'lodash'
+
+//use when only checking if user is admin
+export const withAdminContainer =
+  <Type, ArgsType>(
+    resolver: (_parent: void, args: ArgsType, ctx: Context) => Type,
+    errorMessage: string | undefined = undefined
+  ) =>
+  async (_parent: void, args: ArgsType, ctx: Context) => {
+    const { req } = ctx
+
+    //check if is admin
+    if (errorMessage) {
+      if (!isAdmin(req)) throw new Error(errorMessage)
+    }
+
+    isAdminOrThrow(req)
+
+    return resolver(_parent, args, ctx)
+  }
+
+//use when checking if admin AND if user exists
+export const withAdminUserContainer =
+  <Type, ArgsType>(
+    resolver: (_parent: void, args: ArgsType, ctx: Context) => Type,
+    errorMessage: string | undefined = undefined
+  ) =>
+  async (_parent: void, args: ArgsType, ctx: Context) => {
+    const { req } = ctx
+
+    //check if user exists
+    const authorId = _.get(req, 'user.id')
```

Comment:
Is it possible to use withUserContainer here so we have less duplicate code?

Reply:
One idea i have to is wrap this entire container with withUserContainer

```
export const withAdminUserContainer = withUserContainer(...)
```

Reply:
Hmm looks like this container also has duplicate code with the `withAdminContainer` too -- is it possible to use withAdminContainer here too?

Reply:
Maybe something like this could work
```
withAdminUserContainer = (resolver, errorMessage) => {
  return withUserContainer(WithAdminContainer(resolver, errorMessage))
}
```